chapter_06/dxl/ex3.pl

```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/perl
+use strict;
+use warnings;
+
+my $max_length = 0;
+
+foreach my $key ( keys %ENV ) {
+  my $key_length = length($key);
+  if ( $key_length > $max_length ) {
+    $max_length = $key_length;
+  }
+}
+
+foreach my $key ( sort keys %ENV ) {
+  print $key . " " x ($max_length - length($key)) . "| $ENV{$key}\n";
```

Comment:
1. Nicer to use printf
2. The arithmetic isn't necessary, just negate the number e.g. -$max_length

Reply:
Oh, I didn't know you could use the repeat magic with negative numbers... :hushed: I'll try it, I am not sure I understand how it would work!

Reply:
Apparently `x` doesn't like negative numbers!
```
Negative repeat count does nothing at ex3.pl line 15.
```

Reply:
Shoot. Sorry I meant with printf